src/btree/bt_prefetch.c

```diff
@@ -25,48 +25,43 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
     conn = S2C(session);
     block_preload = 0;
 
-    /*
-     * FIXME-WT-11759 Consider whether we should have these asserts here or swallow up the errors
-     * instead.
-     */
-    WT_ASSERT_ALWAYS(session, F_ISSET(ref, WT_REF_FLAG_LEAF),
-      "Pre-fetch starts with a leaf page and reviews the parent");
+    if ((F_ISSET(ref, WT_REF_FLAG_LEAF)) || (__wt_session_gen(session, WT_GEN_SPLIT) != 0)) {
```

Comment:
Have you considered flipping the direction of the condition here? That way you can get the error handling out of the way earlier (instead of it hiding down the bottom), hoist the main body of the function up a level, and save the reader from having to remember some context.

Reply:
Thanks for the suggestion! Implemented in [f435cdd](https://github.com/wiredtiger/wiredtiger/pull/9949/commits/f435cdd8b675ec4720de444235b85c4c037dc1ef)